app/views/admin/supplier_users/index.html.erb

```diff
@@ -0,0 +1,20 @@
+<%= render "admin/suppliers/layout", locals={page: "users"} %>
+<%= govuk_link_to "Add a new user", new_admin_supplier_user_path, button: true %>
+<table class="govuk-table">
+  <thead>
+    <tr class="govuk-table__row">
+      <th scope="col" class="govuk-table__header">Name</th>
+      <th scope="col" class="govuk-table__header">Email address</th>
+      <th scope="col" class="govuk-table__header">Organisation</th>
+    </tr>
+  </thead>
+  <tbody class="govuk-table__body">
+    <% @users.each do |user| %>
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell"><%= govuk_link_to user.full_name, "#" %></td>
+        <td class="govuk-table__cell"><%= user.email %></td>
+        <td class="govuk-table__cell"><%= user.supplier_name %></td>
```

Comment:
Does that lead to n+1 queries? Or is Rails clever enough to remember it loaded the suppliers before it loaded users (in controller)

Reply:
It does, not sure how else to do it though

Reply:
Specific DTO type thing for the view? How much do we care?

Reply:
I mean, we care a little. I would not say enough for a separate DTO, but maybe enough to examine how we are loading the data. 

If I understand it right, we load the providers, then we load their users, then we sort the users.

I wonder if what we want to do instead is load the users who have the right profiles, along the suppliers they have through the profiles. Then sort them and return them. 

Thoughts?

Reply:
That to me looks something like 
```
User
                     .left_outer_joins(:lead_provider_profile)
                     .left_outer_joins(:delivery_partner_profile)
                     .where.not(lead_provider_profile: { lead_provider_id: nil })
                     .or(User
                           .left_outer_joins(:lead_provider_profile)
                           .left_outer_joins(:delivery_partner_profile)
                           .where.not(delivery_partner_profile: { delivery_partner_id: nil })
                     )
```
Would you agree?

Reply:
Do we need to check lead provider . delivery partner ids not being nil? 

Surely we don't allow provider profiles with no provider?

Reply:
We could also have two queries, one for lead provider users and one for delivery partner users. Better than N+1 :P

Reply:
That doesn't solve the N+1 though

Reply:
I don't know how to do that

Reply:
```
lead_provider_users = User.joins(:lead_provider_profile).includes(:lead_provider)
delivery_partner_users = Users.join(:delivery_partner_profile).includes(:delivery_partner)
<<some code to sort users by full name in ruby>>
```
Is where I would start from.

Reply:
Happy to pair if we somehow find some time when we are free